src/lib/RequestParser.ts

```diff
@@ -214,7 +205,17 @@ export class RequestParser {
                                 case PaymentType.DIRECT:
                                     switch (option.currency) {
                                         case Currency.NIM:
-                                            return new ParsedNimiqDirectPaymentOptions(option, data);
+                                            // Once extraData from MultiCurrencyCheckoutRequest is removed
+                                            // the next few lines become obsolete.
+                                            if (!option.protocolSpecific.extraData) {
+                                                option.protocolSpecific.extraData = checkoutRequest.extraData;
+                                                if (option.protocolSpecific.extraData) {
+                                                    console.warn('Usage of MultiCurrencyCheckoutRequest.extraData is'
+                                                        + ' deprecated. Use NimiqDirectPaymentOptions.protocolSpecific'
+                                                        + '.extraData instead');
+                                                }
+                                            }
```

Comment:
Suggestion to put the warning an indentation level higher:

```js
if (checkoutRequest.extraData) {
    console.warn(...);
}
if (!option.protocolSpecific.extraData) {
    option.protocolSpecific.extraData = checkoutRequest.extraData;
}
```
We could even move the assignment into that first condition, which would always overwrite any data in protocolSpecific with the global extraData (which I think is reasonable).

Reply:
I certainly don't think that the deprecated version should have priority over the new version.

The check could move up, but works just as well as it is.